onmt/modules/BiEncoder.lua

```diff
@@ -42,27 +59,39 @@ Parameters:
   * `rnn` - recurrent template module.
   * `merge` - fwd/bwd merge operation {"concat", "sum"}
 ]]
-function BiEncoder:__init(input, rnn, merge)
+function BiEncoder:__init(args, input)
   parent.__init(self)
 
-  self.fwd = onmt.Encoder.new(input, rnn)
-  self.bwd = onmt.Encoder.new(input:clone('weight', 'bias', 'gradWeight', 'gradBias'), rnn:clone())
+  self.args = onmt.utils.ExtendedCmdLine.getModuleOpts(args, options)
 
-  self.args = {}
-  self.args.merge = merge
+  local orgRNNSize = args.rnn_size
 
-  self.args.rnnSize = rnn.outputSize
-  self.args.numEffectiveLayers = rnn.numEffectiveLayers
+  -- Compute rnn hidden size depending on hidden states merge action.
+  if self.args.brnn_merge == 'concat' then
+    if args.rnn_size % 2 ~= 0 then
+      error('in concat mode, rnn_size must be divisible by 2')
+    end
+    args.rnn_size = args.rnn_size / 2
+  end
+
+  self.args.rnn_size = args.rnn_size
```

Comment:
As said in another PR, renaming:

* `self.args.rnnSize` to `self.args.rnn_size`
* and `self.args.merge` to `self.args.brnn_merge`

is actually breaking existing `BiEncoder` models as the `self.args` table is serialized and used in the forward and backward methods. A workaround would be to add:

```lua
self.args.rnn_size = self.args.rnn_size or self.args.rnnSize
self.args.brnn_merge = self.args.brnn_merge or self.args.merge
```

in the `load` function.